src/components/share-button/share-button.component.ts

```diff
@@ -5,21 +5,19 @@ import {
     Output,
     EventEmitter,
     Renderer,
-    ViewChild,
     ElementRef,
     ChangeDetectionStrategy
 } from '@angular/core';
 
-import { ShareButton, ShareArgs } from '../../helpers/share-buttons.class';
-import { ShareButtonsService } from '../../service/share-buttons.service';
-import { WindowService } from '../../service/window.service';
-import { ShareProvider } from '../../helpers/share-provider.enum';
-
-declare var global: any; // To make AoT compiler (ngc) happy
+import {ShareButton, ShareArgs} from '../../classes/share-buttons.class';
+import {ShareButtonsService} from '../../services/share-buttons.service';
+import {WindowService} from '../../services/window.service';
+import {ShareProvider} from '../../helpers/share-provider.enum';
+import {Helper} from '../../helpers/share.helper';
 
 @Component({
     selector: 'share-button',
-    template: `<button  #btn (click)='share()'></button>`,
+    template: ``,
```

Comment:
Wrong move IMHO! 👎  
I think we should avoid DOM manipulation as much as possible and instead privilege the  Angular Way (using Templates, @ViewChild, ..) which is recommended and offer the advantage of being statically analyzable (AoT) 
for earlier detection of errors.